cpp/dolfinx/fem/interpolate.h

```diff
@@ -640,4 +644,124 @@ void interpolate(Function<T>& u, const Function<T>& v)
   }
 }
 
+/// Interpolate from a dolfinx Expression into a compatible Function on the same
+/// mesh
+/// @param[in, out] u The function to interpolate into
+/// @param[in] expr The expression to be interpolated
+/// @param[in] cells List of cell indices (local to process) to interpolate on
+template <typename T>
+void interpolate(Function<T>& u, const Expression<T>& expr,
```

Comment:
This looks like a lot of duplicated code. I think with some refactoring it will be possible to increase code re-use.

Reply:
Code has now been refactored to re-use interpolate(u, f, cells)